Examples/Sources/MacroExamples/Implementation/Member/CaseDetectionMacro.swift

```diff
@@ -22,29 +22,28 @@ public enum CaseDetectionMacro: MemberMacro {
   ) throws -> [DeclSyntax] {
     declaration.memberBlock.members
       .compactMap { $0.decl.as(EnumCaseDeclSyntax.self) }
-      .map { $0.elements.first!.name }
-      .map { ($0, $0.initialUppercased) }
-      .map { original, uppercased in
+      .map { $0.elements }
+      .flatMap { $0 }
+      .map { $0.name }
+      .map { ($0, $0.pascalcased) }
+      .map { original, pascalcased in
         """
-        var is\(raw: uppercased): Bool {
-          if case .\(raw: original) = self {
-            return true
-          }
-
-          return false
+        var is\(raw: pascalcased): Bool {
+            self == .\(raw: original)
         }
         """
```

Comment:
Would it be easier to just unify these three `map` calls into one? Also, I think we were using `if case` instead of `==` here so that this also works for enums with associated values.

```swift
.map { 
  """
  var is\(raw: $0.name.asCamelCase): Bool {
    if case .\(raw: $0.name) = self {
      return true
    }
    return false
  }
  """
}
```